### PR TITLE
MODCON-56 implemented get publication results

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
https://issues.folio.org/browse/MODCON-56

## Approach
* implemented get publication results

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/swagger.api/publications.yaml [hunk 1]
```diff
@@ -39,6 +39,20 @@ paths:
           $ref: '#/components/responses/BadRequest'
         '500':
           $ref: '#/components/responses/InternalServerError'
+  /publications/{publicationId}/results:
+    get:
+      summary: Get publication details
```
Comment (SerhiiNosko) [inline src/main/resources/swagger.api/publications.yaml:44]: should be: Get publication results

**Review by SerhiiNosko — APPROVED** [review]

File: src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java [hunk 2]
```diff
@@ -265,4 +267,28 @@ private PublicationResponse buildPublicationResponse(UUID publicationId) {
...
+  @Override
+  public PublicationResultCollection getPublicationResults(UUID consortiumId, UUID publicationId){
+    log.debug("getPublicationResults:: Trying to retrieve publication results by consortiumId: {} and publicationId id: {}", consortiumId, publicationId);
+
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
+    var publicationStatusEntity = publicationStatusRepository.findById(publicationId)
+      .orElseThrow(() -> new ResourceNotFoundException("publicationId", String.valueOf(publicationId)));
+
+    var ptrEntities = publicationTenantRequestRepository.findByPcStateId(publicationId, PageRequest.of(0, Integer.MAX_VALUE));
+    log.info("getPublicationResults:: Found {} of {} expected tenant request records", ptrEntities.getTotalElements(), publicationStatusEntity.getTotalRecords());
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java:280]: maybe also log pblication and consortium ids to make this info entry more useful?

File: src/main/resources/swagger.api/schemas/publication_schema.yaml [hunk 3]
```diff
@@ -72,3 +72,29 @@ PublicationStatusError:
...
+      type: array
+      description: "The list of publication results"
+      items:
+        type: object
+        $ref: "publication_schema.yaml#/PublicationResult"
+    totalRecords:
+      type: integer
+
+PublicationResult:
+  description: publication result
```
Comment (tatsiana-tarhonskaya) [inline src/main/resources/swagger.api/schemas/publication_schema.yaml:90]: there's usage of quotes in part of places and in others there's not, we should pick some common approach

**Review by tatsiana-tarhonskaya — APPROVED** [review]

File: src/main/resources/swagger.api/schemas/publication_schema.yaml (hunk 3 above)
Comment (grabsefx) [reply src/main/resources/swagger.api/schemas/publication_schema.yaml:90]: https://issues.folio.org/browse/MODCON-76
created a separate ticket where we can decide how to deal with such issues

**Pull request merged**